core/src/main/java/org/apache/druid/common/exception/SanitizableException.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.common.exception;
+
+import java.util.function.Function;
+
+public interface SanitizableException
+{
+  /**
+   * Apply the function for transforming the error message then
+   * return new Exception with sanitized fields and transformed message.
+   */
+  Exception transform(
```

Comment:
nit: rename to `sanitize` to be inline with `SanitizableException`

Reply:
heh oops, was my suggestion to rename `transform`, but I guess I also suggested renaming `SanitizableException` to something else, though I feel less strongly about the name of this than the property name in the config, (which I think should be `druid.server.http.enableErrorResponseTransform` and `druid.server.http.errorResponseTransformStrategy`)

Reply:
Done